android/composeui/src/main/java/com/stadiamaps/ferrostar/composeui/views/InstructionsView.kt

```diff
@@ -64,60 +67,78 @@ fun InstructionsView(
 
           // Expanded content
-          if (isExpanded && remainingSteps != null && remainingSteps.count() > 1) {
-            item { HorizontalDivider(thickness = 1.dp) }
-            items(remainingSteps.drop(1)) { step ->
-              step.visualInstructions.firstOrNull()?.let { upcomingInstruction ->
-                Spacer(modifier = Modifier.height(8.dp))
-                ManeuverInstructionView(
-                    text = upcomingInstruction.primaryContent.text,
-                    distanceFormatter = distanceFormatter,
-                    distanceToNextManeuver = step.distance,
-                    theme = theme) {
-                      contentBuilder(upcomingInstruction)
+          val showMultipleRows = isExpanded && remainingSteps != null && remainingSteps.count() > 1
+          if (showMultipleRows) {
+            Spacer(modifier = Modifier.height(8.dp))
+            HorizontalDivider(thickness = 1.dp)
+            Spacer(modifier = Modifier.height(8.dp))
+          }
+
+          if (isExpanded) {
+            Box(modifier = Modifier.weight(1f)) {
+              LazyColumn(
+                modifier = Modifier.fillMaxSize(),
+                verticalArrangement = Arrangement.spacedBy(8.dp)
+              ) {
+                if (remainingSteps != null) {
+                  items(remainingSteps.drop(1)) { step ->
+                    step.visualInstructions.firstOrNull()?.let { upcomingInstruction ->
+                      ManeuverInstructionView(
+                        text = upcomingInstruction.primaryContent.text,
+                        distanceFormatter = distanceFormatter,
+                        distanceToNextManeuver = step.distance,
+                        theme = theme
+                      ) {
+                        contentBuilder(upcomingInstruction)
+                      }
+                      Spacer(modifier = Modifier.height(8.dp))
+                      HorizontalDivider(thickness = 1.dp)
                     }
-                Spacer(modifier = Modifier.height(16.dp))
-                HorizontalDivider(thickness = 1.dp)
+                  }
+                }
               }
             }
           }
-        }
 
-        if (isExpanded) {
-          Spacer(modifier = Modifier.weight(1.0f))
+          if (showMultipleRows) {
+            Spacer(modifier = Modifier.height(16.dp))
+          }
         }
 
         PillDragHandle(
             isExpanded,
             // The modifier here lets us keep the container as slim as possible
-            modifier = Modifier.offset(y = 4.dp).align(Alignment.CenterHorizontally),
+            modifier =
+                Modifier.offset {
+                      IntOffset(
+                          0,
+                          if (isExpanded) {
+                            (-4 * density).toInt()
+                          } else {
+                            (-8 * density).toInt()
+                          })
+                    }
```

Comment:
Unfortunately the new layout structure required some changes here. Android Studio gave a lint warning that I should use this closure version of offset since it depends on state. TIL.